src/freenet/client/filter/ElementInfo.java

```diff
@@ -218,7 +227,27 @@ public class ElementInfo {
 			"first-letter",
 			"before",
 			"after",
-			"target"
+			"target",
+			"any-link",
+			"default",
+			"defined",
+			"disabled",
+			"empty",
+			"enabled",
+			"focus-visible",
+			"dir",
```

Comment:
is dir necessary here? We already have a dedicated parsing rule for it. If it is: please add a test that fails without your change.

Reply:
dir(ltr) and dir(rtl) as pseudoclasses, added here.